main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DateTripsValidator.java

```diff
@@ -0,0 +1,109 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.SortedSet;
+import javax.inject.Inject;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.input.CurrentDateTime;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.SeverityLevel;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.*;
+import org.mobilitydata.gtfsvalidator.type.GtfsDate;
+import org.mobilitydata.gtfsvalidator.util.CalendarUtil;
+import org.mobilitydata.gtfsvalidator.util.TripCalendarUtil;
+
+/**
+ * Validates that trip data exists for the next 7 days. Dates with the majority trips per day should
+ * be included for at least the next 7 day period. see
+ * https://github.com/MobilityData/gtfs-validator/issues/886#issuecomment-832237225
+ *
+ * <p>Generated notices:
+ *
+ * <ul>
+ *   <li>{@link TripDataShouldBeValidForNext7DaysNotice}.
+ * </ul>
+ */
+@GtfsValidator
+public class DateTripsValidator extends FileValidator {
+
+  private final GtfsCalendarDateTableContainer calendarDateTable;
+  private final GtfsCalendarTableContainer calendarTable;
+  private final GtfsTripTableContainer tripContainer;
+  private final GtfsFrequencyTableContainer frequencyTable;
+  private final CurrentDateTime currentDateTime;
+
+  @Inject
+  DateTripsValidator(
+      CurrentDateTime currentDateTime,
+      GtfsCalendarDateTableContainer calendarDateTable,
+      GtfsCalendarTableContainer calendarTable,
+      GtfsTripTableContainer tripContainer,
+      GtfsFrequencyTableContainer frequencyTable) {
+    this.currentDateTime = currentDateTime;
+    this.calendarTable = calendarTable;
+    this.calendarDateTable = calendarDateTable;
+    this.tripContainer = tripContainer;
+    this.frequencyTable = frequencyTable;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+
+    LocalDate now = currentDateTime.getNow().toLocalDate();
+
+    final Map<String, SortedSet<LocalDate>> servicePeriodMap =
+        CalendarUtil.servicePeriodToServiceDatesMap(
+            CalendarUtil.buildServicePeriodMap(calendarTable, calendarDateTable));
+
+    var tripCounts =
+        TripCalendarUtil.countTripsForEachServiceDate(
+            servicePeriodMap, tripContainer, frequencyTable);
+    var majorityServiceDates = TripCalendarUtil.computeMajorityServiceCoverage(tripCounts);
+    LocalDate currentDatePlusSevenDays = now.plusDays(7);
+
+    if (majorityServiceDates != null) {
```

Comment:
`majorityServiceDates` cannot be null, but it can be `Optional.empty`.

Reply:
Good catch - my mistake.